sources/packages/web/src/components/students/applicationTracker/ApplicationProgressBar.vue

```diff
@@ -85,26 +80,68 @@ export default defineComponent({
       type: String as PropType<ApplicationStatus>,
       required: true,
     },
-    statusUpdatedOn: {
-      type: String,
-      required: true,
-    },
   },
   setup(props) {
     const hasDeclinedCard = ref(false);
-    const applicationStatusTracker = ref<Record<number, ApplicationStatus>>({
+    const applicationTrackerLabels = ref<Record<number, ApplicationStatus>>({
       0: ApplicationStatus.submitted,
       1: ApplicationStatus.inProgress,
       2: ApplicationStatus.assessment,
       3: ApplicationStatus.enrollment,
       4: ApplicationStatus.completed,
     });
-    // trackFillColor will vary when more status are added.
-    const trackFillColor = ref("warning");
+    const applicationProgressDetails = ref(
+      {} as ApplicationProgressDetailsAPIOutDTO,
+    );
+    const applicationEndStatus = ref({} as ApplicationEndStatusIconDetails);
+    const trackFillColor = computed(() => {
+      if (props.applicationStatus === ApplicationStatus.completed) {
+        return "success";
+      }
+      if (applicationEndStatus.value.endStatusType === "error") {
+        return "error";
+      }
+      return "warning";
+    });
+
+    onMounted(async () => {
+      applicationProgressDetails.value =
+        await ApplicationService.shared.getApplicationProgressDetails(
```

Comment:
Since we have a specific API method to get the application progress (`getApplicationProgressDetails`) I would not mind having the API returning a status for it that would make all the logic from line 112 till 140 more simple.

Reply:
I was initially thinking the same, then with the feeling that the logic is a client specific one, moved the logic to client side keeping API simple.

Reply:
"client specific one" is a debatable point. Unless the API is returning some property like `statusForTheProgressBar` or `colorToThatComponent`, having the API returning something aligned with the `ProgressDetails` will not be a problem IMO.

Reply:
As discussed leaving the logic to get the end status stay at the client side and when we get the requirement to show the application end status we can either have the API return some status or move the existing client side code to composable.